src/mono/mono/metadata/assembly-internals.h

```diff
@@ -141,6 +141,30 @@ mono_assembly_get_image_internal (MonoAssembly *assembly);
 void
 mono_set_assemblies_path_direct (char **path);
 
+typedef enum {
+	BUNDLED_BLOB,
+	BUNDLED_ASSEMBLY,
+	BUNDLED_SATELLITE_ASSEMBLY,
+	BUNDLED_BINARY,
+	BUNDLED_RESOURCE_COUNT,
+} BundledResourceType;
+
+typedef struct BundledResource {
+	const char *culture; // Satellite assemblies
```

Comment:
Maybe it is nicer if you add base struct types and had type as first parameter, data and size, maybe name and then you do a different struct with the culture and then you pass the struct to the mono_add_bundle_resource and we have an API that we can extend with different types going forward. I also think the API should own copies of name and culture strings, the data parameter will of course be owned by caller since it might be a large memory region, potentially backed by memory mapped file or similar. Right now we strdup these strings in MonoBundledAssembly but maybe they should be owned by the BundleResource struct variations.

Reply:
Implemented the suggested generalized MonoBundled*Resource structs with symfile information folded into it.